src/EventListener/CreateSchemaListener.php

```diff
@@ -96,11 +95,15 @@ public function postGenerateSchemaTable(GenerateSchemaTableEventArgs $eventArgs)
         $revisionTable->addIndex([$this->config->getRevisionFieldName()], $revIndexName);
         $revisionForeignKeyName = $this->config->getRevisionFieldName().'_'.md5($revisionTable->getName()).'_fk';
 
-        $foreignColumnNames = $revisionsTable->getPrimaryKeyColumns();
-
         // TODO: Use always array_keys when dropping support for DBAL 2
-        if (!interface_exists(Result::class)) {
```

Comment:
This was buggy.

The array key is needed for DBAL 3, but the interface_exist return false for both DBAL 3 and 2 (not sure why).

This seems safer to check the return value instead.

Reply:
Is the comment with the TODO relevant or should we move/change it somewhere else?

Reply:
It's relevant.

DBAL 2 return string[]
DBAL 3 return Column[]

So currently I support both, but when DBAL 2 is dropped we can use array_keys without any more questions.